### PR TITLE
[Group refactoring] Relationship model

### PR DESCRIPTION
**Pull request by bgeuken**

### CODE REVIEW

File: src/api/app/mixins/has_relationships.rb [hunk 1]
```diff
@@ -204,13 +204,14 @@ def update_generic_relationships(xmlhash)
...
-          record = self.relationships.build(role: role)
+          record = self.relationships.new(role: role)
           @updater.set_item(record, item)
           pcache[role.title] = :new
         end
       else
-        record = self.relationships.build(role: role)
+        record = self.relationships.new(role: role)
         @updater.set_item(record, item)
+        record.save
```
Comment (hennevogel) [inline src/api/app/mixins/has_relationships.rb]: Hm why is that needed?

Comment (bgeuken) [inline src/api/app/mixins/has_relationships.rb]: build is just an alias for new.

Comment (bgeuken) [inline src/api/app/mixins/has_relationships.rb]: Afaik we did that for some test

Comment (bgeuken) [inline src/api/app/mixins/has_relationships.rb]: @hennevogel So tests would keep running without the save. I gone remove it.

Comment (hennevogel) [inline src/api/app/mixins/has_relationships.rb]: okay

**Comment by bgeuken** [comment]
*rebased*

Please review again.

**Comment by hennevogel** [comment]
Looks good to me!
